utils/TableGen/AsmWriterEmitter.cpp

```diff
@@ -736,18 +736,21 @@ class IAPrinter {
     O.indent(4) << '}';
   }
 
-  bool operator==(const IAPrinter &RHS) {
+  // HLSL Change begin
+  bool operator==(const IAPrinter &RHS) const {
     if (Conds.size() != RHS.Conds.size())
       return false;
 
     unsigned Idx = 0;
-    for (std::vector<std::string>::iterator
-           I = Conds.begin(), E = Conds.end(); I != E; ++I)
+    for (std::vector<std::string>::const_iterator I = Conds.begin(),
```

Comment:
Much as I personally find the `auto` keyword annoying, I'd prefer that we keep things as consistent with upstream LLVM as possible and [this change](https://github.com/llvm/llvm-project/commit/4ab57cd9abb9535149e62c18461b56b94b09285a) opted to just used `auto` instead of specifying a const_iterator. I don't think it's necessary to take the other aspects of that change though. By making it consistent with upstream, you wouldn't need the "HLSL Change" comments either

Reply:
Thanks, I updated to align with LLVM change.